core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1933,29 +1951,45 @@ private[spark] class DAGScheduler(
       maybeEpoch = None)
   }
 
+  /**
+   * Handles removing an executor from the BlockManagerMaster as well as unregistering shuffle
+   * outputs for the executor or optionally its host.
+   *
+   * @param execId executor to be removed
+   * @param fileLost If true, indicates that we assume we've lost all shuffle blocks associated
+   *   with the executor; this happens if the executor serves its own blocks (i.e., we're not
+   *   using external shuffle), the Standalone worker (which serves the shuffle data) is lost,
```

Comment:
`external shuffle` -> `external shuffle service`

Reply:
Other clusters, like YARN, also could use external shuffle service. So they may face the same issue.

Reply:
I'm not sure I understand the point you're making here. The 3 conditions when `fileLost=true` are: (1) we're not using the external shuffle service (`!env.blockManager.externalShuffleServiceEnabled`); or (2) the Standalone worker is lost (as I explained above); or (3) a FetchFailed occurred.
In the case of case of YARN + external shuffle service, `fileLost=true` happens only with a FetchFailed.